controllers/cluster_controller.go

```diff
@@ -129,6 +129,7 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager, log logr.Logger)
 // +kubebuilder:rbac:groups=distro.eks.amazonaws.com,resources=releases,verbs=get;list;watch
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awssnowclusters;awssnowmachinetemplates;awssnowippools;vsphereclusters;vspheremachinetemplates;dockerclusters;dockermachinetemplates;tinkerbellclusters;tinkerbellmachinetemplates,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",namespace=eksa-system,resources=secrets,verbs=delete;
+// +kubebuilder:rbac:groups=tinkerbell.org,resources=hardware;hardware/status,verbs=get;list;watch;update;patch
```

Comment:
Also do we need to give it bmc permissions since the catalog reads that as well?

Reply:
> Also do we need to give it bmc permissions since the catalog reads that as well?

I was planning on a separate PR for this. With the BMC checks included.

Reply:
> should we restrict this to hardware objects only in the `eksa-system` ns?

I think we could? We do always expect hardware to be in the eksa namesapce.